captum/attr/_core/layer/layer_integrated_gradients.py

```diff
@@ -0,0 +1,150 @@
+    _format_attributions,
+    _format_input_baseline,
+)
+
+from captum.attr._utils.gradient import _forward_layer_eval
+
+from captum.attr._utils.attribution import LayerAttribution
+from captum.attr._core.integrated_gradients import IntegratedGradients
+from captum.attr._utils.gradient import _run_forward
+
+
+class LayerIntegratedGradients(LayerAttribution, IntegratedGradients):
+    def __init__(self, forward_func, layer, device_ids=None):
+        r"""
+        Args:
+            forward_func (callable):  The forward function of the model or any
+                          modification of it
+        """
+        LayerAttribution.__init__(self, forward_func, layer, device_ids=device_ids)
+        IntegratedGradients.__init__(self, forward_func)
+
+    def attribute(
+        self,
+        inputs,
+        baselines=None,
+        target=None,
+        additional_forward_args=None,
+        n_steps=50,
+        method="gausslegendre",
+        internal_batch_size=None,
+        return_convergence_delta=False,
+        attribute_to_layer_input=False,
+    ):
+        inps, baselines = _format_input_baseline(inputs, baselines)
+        _validate_input(inps, baselines, n_steps, method)
+
+        baselines = _tensorize_baseline(inps, baselines)
+        additional_forward_args = _format_additional_forward_args(
+            additional_forward_args
+        )
+
+        if self.device_ids is None:
+            self.device_ids = getattr(self.forward_func, "device_ids", None)
+        inputs_layer = _forward_layer_eval(
+            self.forward_func,
+            inps,
+            self.layer,
+            device_ids=self.device_ids,
+            additional_forward_args=additional_forward_args,
+            attribute_to_layer_input=attribute_to_layer_input,
+        )
+
+        baselines_layer = _forward_layer_eval(
+            self.forward_func,
+            baselines,
+            self.layer,
+            device_ids=self.device_ids,
+            additional_forward_args=additional_forward_args,
+            attribute_to_layer_input=attribute_to_layer_input,
+        )
+
+        # inputs -> these inputs are scaled
+        def gradient_func(
+            forward_fn, inputs, target_ind=None, additional_forward_args=None
+        ):
+            if self.device_ids is None:
+                scattered_inputs = inputs
+            else:
+                scattered_inputs = scatter(inputs, target_gpus=self.device_ids)
+
+            scattered_inputs_dict = {
+                scattered_input[0].device: scattered_input
+                for scattered_input in scattered_inputs
+            }
+
+            with torch.autograd.set_grad_enabled(True):
+
+                def layer_forward_hook(module, hook_inputs, hook_outputs=None):
+                    result = scattered_inputs_dict[hook_inputs[0].device]
+                    return (
+                        result[0]
+                        if isinstance(result, tuple) and hook_outputs is not None
```

Comment:
With the current implementation of _forward_layer_eval always returning a tensor, will result ever be a tuple? We will also probably need to update this condition when the layer tuple code merges to return the tuple?

Reply:
Oh, no, that was an issue coming from scatter. Fixed it